src/bcd/bcd/IDenoiser.h

```diff
@@ -13,15 +13,29 @@
 #ifndef I_DENOISER_H
 #define I_DENOISER_H
 
-#include <functional>
-#include <memory>
 
 namespace bcd
 {
 
 template<class T>
 class DeepImage;
 
+class IProgressReporter
+{
+  public:
+    IProgressReporter();
+    virtual ~IProgressReporter();
+
+    virtual void progress(const float i_progress) const = 0;
```

Comment:
What does the `i_` prefix mean? Input? Is that a convention of the BCD code?

Reply:
input param. I kept the original style of the code.